Crash when user selects audio that was just recorded

<!-- 

Thank you for taking the time to report an ODK Collect issue!

This space is for submitting problems and feature requests. For general usage or form design questions, please email opendatakit@googlegroups.com or for ODK Collect source code questions, please email opendatakit-developers@googlegroups.com

Before filling this form, visit https://github.com/opendatakit/collect/issues?q=is%3Aissue and search to see whether your issue was already reported or fixed. If you find a match, comment on it or add a +1 rather than posting a new issue. If you find a problem you know how to fix, submit a pull request. 🎉

For all problem reports, please use the template below. Also include any relevant stack traces or error messages.

For feature requests, please include the problem description (what problem do you have that can't currently be solved?) and a proposed solution if you have one in mind (optional). You can delete the template. 

-->

#### Software and hardware versions 
Collect v1.4.16, Android v4.4.4, Motorola XT1022

#### Problem description

Null pointer returned on selecting recorded audio.

#### Steps to reproduce the problem

1. Go to the audio widget.
2. Record an audio clip.
3. Click on choose button and select the audio clip that you just recorded. 

#### Expected behavior

The Uri of the chosen audio file should be returned.

#### Other information 

The issue is pretty much the same as  #691.

I got 
```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=3, result=-1, data=Intent { dat=content://com.sonymobile.androidapp.audiorecorder.content/audio/1 typ=audio/raw }} to activity {org.odk.collect.android/org.odk.collect.android.activities.FormEntryActivity}: java.lang.UnsupportedOperationException: You cannot use this method
	at android.app.ActivityThread.deliverResults(ActivityThread.java:3699)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742)
	at android.app.ActivityThread.-wrap16(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.UnsupportedOperationException: You cannot use this method
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:167)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
	at android.content.ContentProviderProxy.delete(ContentProviderNative.java:544)
	at android.content.ContentResolver.delete(ContentResolver.java:1327)
	at org.odk.collect.android.activities.FormEntryActivity.onActivityResult(FormEntryActivity.java:713)
	at android.app.Activity.dispatchActivityResult(Activity.java:6456)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:3695)
	... 9 more
   Force finishing activity org.odk.collect.android/.activities.FormEntryActivity
```
filed as #705

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
@@ -233,10 +233,7 @@ public class AudioWidget extends QuestionWidget implements IBinaryWidget {
 
     @Override
     public void setBinaryData(Object binaryuri) {
-        // when replacing an answer. remove the current media.
-        if (mBinaryName != null) {
-            deleteMedia();
-        }
+
 
         // get the file path and create a copy in the instance folder
         String binaryPath = MediaUtils.getPathFromUri(this.getContext(), (Uri) binaryuri,
@@ -260,6 +257,10 @@ public class AudioWidget extends QuestionWidget implements IBinaryWidget {
             Uri AudioURI = getContext().getContentResolver().insert(
                     Audio.Media.EXTERNAL_CONTENT_URI, values);
             Log.i(t, "Inserting AUDIO returned uri = " + AudioURI.toString());
+            // when replacing an answer. remove the current media.
+            if (mBinaryName != null && !mBinaryName.equals(newAudio.getName())) {
+                deleteMedia();
+            }
             mBinaryName = newAudio.getName();
             Log.i(t, "Setting current answer to " + newAudio.getName());
         } else {

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/AudioWidget.java
@@ -234,7 +234,6 @@ public class AudioWidget extends QuestionWidget implements IBinaryWidget {
     @Override
     public void setBinaryData(Object binaryuri) {
 
-
         // get the file path and create a copy in the instance folder
         String binaryPath = MediaUtils.getPathFromUri(this.getContext(), (Uri) binaryuri,
                 Audio.Media.DATA);

